### PR TITLE
Switch to pathlib

### PR DESCRIPTION
**Pull request by lukemassa**
Switching code from using os.path to using Pathlib (per request in https://github.com/hjacobs/pykube/pull/55)

### CODE REVIEW

**Comment by lukemassa** [comment]
@hjacobs Marked this as WIP as I clean up the tests, I just wanted to make sure this is what you were looking for?

Basically I found once I started replacing some references it made sense to carry all the way through, that is to only interact with Paths and not strings for as long as I can.

File: tests/test_config.py [hunk 1]
```diff
@@ -12,10 +12,9 @@
 from pykube import exceptions
 
 
-GOOD_CONFIG_FILE_PATH = os.path.sep.join(["tests", "test_config.yaml"])
-DEFAULTUSER_CONFIG_FILE_PATH = os.path.sep.join(
-    ["tests", "test_config_default_user.yaml"]
-)
+BASEDIR = Path("tests")
+GOOD_CONFIG_FILE_PATH = BASEDIR.joinpath("test_config.yaml")
```
Comment (hjacobs) [inline tests/test_config.py]: you can use the short form with the `/` operator:

```python
BASEDIR / "test_config.yaml"
```

Comment (lukemassa) [reply tests/test_config.py]: slick!

**Comment by lukemassa** [comment]
Apologies for the delay here, this is now finished:
```
luke@Lukes-MacBook-Air pykube % grep -r 'os\.path' pykube tests
luke@Lukes-MacBook-Air pykube % 
```

**Pull request merged**